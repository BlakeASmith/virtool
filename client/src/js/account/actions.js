/**
 *
 *
 * @copyright 2017 Government of Canada
 * @license MIT
 * @author igboyes
 *
 */
import {
    WS_UPDATE_ACCOUNT,
    GET_ACCOUNT,
    UPDATE_ACCOUNT_SETTINGS,
    CREATE_API_KEY,
    UPDATE_API_KEY,
    REMOVE_API_KEY,
    LOGOUT
} from "../actionTypes";


export function wsUpdateAccount (data) {
    return {
        type: WS_UPDATE_ACCOUNT,
        data
    };
}

export function getAccount () {
    return {
        type: GET_ACCOUNT.REQUESTED
    }
}

export function updateAccountSettings (update) {
    return {
        type: UPDATE_ACCOUNT_SETTINGS.REQUESTED,
        update
    };
}

export function createAPIKey (name, permissions, callback) {
    return {
        type: CREATE_API_KEY.REQUESTED,
        name,
        permissions,
        callback
    };
}

export function updateAPIKey (keyId, permissions) {
    return {
        type: UPDATE_API_KEY.REQUESTED,
        keyId,
        permissions
    };
}

export function removeAPIKey (keyId) {
    return {
        type: REMOVE_API_KEY.REQUESTED,
        keyId
    };
}

export function logout () {
    return {
        type: LOGOUT.REQUESTED
    }
}
