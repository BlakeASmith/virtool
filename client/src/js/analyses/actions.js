import {
    WS_INSERT_ANALYSIS,
    WS_UPDATE_ANALYSIS,
    WS_REMOVE_ANALYSIS,
    FIND_ANALYSES,
    GET_ANALYSIS,
    ANALYZE,
    BLAST_NUVS,
    REMOVE_ANALYSIS,
    CLEAR_ANALYSIS,
    COLLAPSE_ANALYSIS,
    SET_PATHOSCOPE_FILTER,
    TOGGLE_SORT_PATHOSCOPE_DESCENDING,
    TOGGLE_SHOW_PATHOSCOPE_READS,
    SET_ACTIVE_HIT_ID,
    TOGGLE_FILTER_ORFS,
    TOGGLE_FILTER_SEQUENCES,
    SET_SEARCH_IDS,
    SET_ANALYSIS_SORT_KEY,
    TOGGLE_RESULT_EXPANDED,
    CLEAR_ANALYSES
} from "../app/actionTypes";
import { simpleActionCreator } from "../utils/utils";

export const wsInsertAnalysis = data => ({
    type: WS_INSERT_ANALYSIS,
    data
});

/**
 * Returns an action that should be dispatched when a analysis document is updated via websocket.
 *
 * @func
 * @param data {object} update data passed in the websocket message
 * @returns {object}
 */
export const wsUpdateAnalysis = data => ({
    type: WS_UPDATE_ANALYSIS,
    data
});

/**
 * Returns an action that should be dispatched when a analysis document is removed via websocket.
 *
 * @func
 * @param data {string} the id for the specific analysis
 * @returns {object}
 */
export const wsRemoveAnalysis = data => ({
    type: WS_REMOVE_ANALYSIS,
    data
});

export const collapseAnalysis = simpleActionCreator(COLLAPSE_ANALYSIS);

export const setActiveHitId = id => ({
    type: SET_ACTIVE_HIT_ID,
    id
});

export const setSearchIds = ids => ({
    type: SET_SEARCH_IDS,
    ids
});

export const setAnalysisSortKey = sortKey => ({
    type: SET_ANALYSIS_SORT_KEY,
    sortKey
});

export const setPathoscopeFilter = key => ({
    type: SET_PATHOSCOPE_FILTER,
    key
});

export const toggleFilterORFs = simpleActionCreator(TOGGLE_FILTER_ORFS);

export const toggleFilterSequences = simpleActionCreator(TOGGLE_FILTER_SEQUENCES);

export const togglePathoscopeSortDescending = simpleActionCreator(TOGGLE_SORT_PATHOSCOPE_DESCENDING);

export const toggleResultExpanded = id => ({
    type: TOGGLE_RESULT_EXPANDED,
    id
});

export const toggleShowPathoscopeReads = simpleActionCreator(TOGGLE_SHOW_PATHOSCOPE_READS);

export const findAnalyses = (sampleId, term, page) => ({
    type: FIND_ANALYSES.REQUESTED,
    sampleId,
    term,
    page
});

/**
 * Returns action that can trigger an API call for retrieving a specific analysis.
 *
 * @func
 * @param analysisId {string} unique analysis id
 * @returns {object}
 */
export const getAnalysis = analysisId => ({
    type: GET_ANALYSIS.REQUESTED,
    analysisId
});

export const clearAnalyses = simpleActionCreator(CLEAR_ANALYSES);
export const clearAnalysis = simpleActionCreator(CLEAR_ANALYSIS);

/**
 * Returns action that can trigger an API call for sample analysis.
 *
 * @func
 * @param sampleId {string} unique sample id
 * @param refId {string} unique id for a reference
 * @param algorithm {string} algorithm type
 * @param userId {string} the id of the requesting user
 * @returns {object}
 */
export const analyze = (sampleId, refId, algorithm, subtractionId, userId) => ({
    type: ANALYZE.REQUESTED,
    algorithm,
    refId,
    sampleId,
    subtractionId,
    userId
});

/**
 * Returns action that can trigger an API call for BLASTing NuV analysis contigs.
 *
 * @func
 * @param analysisId {string} unique analysis id
 * @param sequenceIndex {number} index of the sequence
 * @returns {object}
 */
export const blastNuvs = (analysisId, sequenceIndex) => ({
    type: BLAST_NUVS.REQUESTED,
    analysisId,
    sequenceIndex
});

/**
 * Returns action that can trigger an API call for removing an analysis.
 *
 * @func
 * @param analysisId {string} unique analysis id
 * @returns {object}
 */
export const removeAnalysis = analysisId => ({
    type: REMOVE_ANALYSIS.REQUESTED,
    analysisId
});
