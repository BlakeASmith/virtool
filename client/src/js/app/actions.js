import { PUSH_STATE } from "./actionTypes";

export const pushState = state => ({
    type: PUSH_STATE,
    state
});
