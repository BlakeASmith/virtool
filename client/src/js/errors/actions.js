/**
 * Redux action creator for clearing errors.
 *
 * @module error/actions
 */
import { CLEAR_ERROR } from "../actionTypes";

/**
 * Clears specific temprorarily stored error.
 *
 * @func
 * @returns {object}
 */
export const clearError = (error) => ({
    type: CLEAR_ERROR,
    error
});
