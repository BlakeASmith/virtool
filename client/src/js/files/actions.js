/**
 * Redux action creators for use with file management.
 *
 * @module files/actions
 */

import { simpleActionCreator } from "../utils";
import {
    WS_UPDATE_FILE,
    WS_REMOVE_FILE,
    FIND_FILES,
    REMOVE_FILE,
    UPLOAD,
    UPLOAD_PROGRESS,
    HIDE_UPLOAD_OVERLAY
} from "../actionTypes";

/**
 * Returns an action that should be dispatched when a file document is updated via websocket.
 *
 * @func
 * @param data {object} the data passed in the websocket message
 * @returns {object}
 */
export const wsUpdateFile = (data) => ({
    type: WS_UPDATE_FILE,
    data
});

/**
 * Returns an action that should be dispatched when a file document is removed via websocket.
 *
 * @func
 * @param data {object} the data passed in the websocket message
 * @returns {object}
 */
export const wsRemoveFile = (data) => ({
    type: WS_REMOVE_FILE,
    data
});

/**
 * Returns an action that can trigger an API call that finds file documents.
 *
 * @func
 * @param fileType {string} a file type to filter the returned documents by
 * @param page {number} which page of results to return
 * @returns {object}
 */
export const findFiles = (fileType, page) => ({
    type: FIND_FILES.REQUESTED,
    fileType,
    page
});

/**
 * Returns an action that can trigger the upload of a file to the server.
 *
 * @param localId
 * @param file
 * @param fileType
 * @param onProgress
 * @returns {{type: string, localId: *, file: *, fileType: *, onProgress: *}}
 */
export const upload = (localId, file, fileType, onProgress) => ({
    type: UPLOAD.REQUESTED,
    localId,
    file,
    fileType,
    onProgress
});

export const removeFile = (fileId) => ({
    type: REMOVE_FILE.REQUESTED,
    fileId
});

export const uploadProgress = (localId, progress) => ({
    type: UPLOAD_PROGRESS,
    localId,
    progress
});

export const hideUploadOverlay = simpleActionCreator(HIDE_UPLOAD_OVERLAY);
