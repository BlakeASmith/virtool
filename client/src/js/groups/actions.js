import { simpleActionCreator } from "../utils";
import { LIST_GROUPS, CREATE_GROUP, SET_GROUP_PERMISSION, REMOVE_GROUP } from "../actionTypes";

export const listGroups = simpleActionCreator(LIST_GROUPS.REQUESTED);

export const createGroup = (groupId) => ({
    type: CREATE_GROUP.REQUESTED,
    groupId
});

export const setGroupPermission = (groupId, permission, value) => ({
    type: SET_GROUP_PERMISSION.REQUESTED,
    groupId,
    permission,
    value
});

export const removeGroup = (groupId) => ({
    type: REMOVE_GROUP.REQUESTED,
    groupId
});
