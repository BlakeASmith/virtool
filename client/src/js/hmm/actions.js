/**
 *
 *
 * @copyright 2017 Government of Canada
 * @license MIT
 * @author igboyes
 *
 */
import { FIND_HMMS, GET_HMM, INSTALL_HMMS } from "../actionTypes";

export const findHMMs = (term, page) => {
    return {
        type: FIND_HMMS.REQUESTED,
        term,
        page
    };
};

export const getHmm = (hmmId) => {
    return {
        type: GET_HMM.REQUESTED,
        hmmId: hmmId
    };
};

export const installHMMs = () => {
    return {
        type: INSTALL_HMMS.REQUESTED
    }
};
