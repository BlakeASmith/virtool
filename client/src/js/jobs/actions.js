import { simpleActionCreator } from "../utils";
import {
    WS_UPDATE_JOB,
    WS_REMOVE_JOB,
    FIND_JOBS,
    GET_JOB,
    CANCEL_JOB,
    REMOVE_JOB,
    CLEAR_JOBS,
    GET_RESOURCES
} from "../actionTypes";

export const wsUpdateJob = (data) => ({
    type: WS_UPDATE_JOB,
    data
});

export const wsRemoveJob = (jobId) => ({
    type: WS_REMOVE_JOB,
    jobId
});

export const findJobs = (term, page) => ({
    type: FIND_JOBS.REQUESTED,
    term,
    page
});

export const getJob = (jobId) => ({
    type: GET_JOB.REQUESTED,
    jobId
});

export const cancelJob = (jobId) => ({
    type: CANCEL_JOB.REQUESTED,
    jobId
});

export const removeJob = (jobId) => ({
    type: REMOVE_JOB.REQUESTED,
    jobId
});

export const clearJobs = (scope) => ({
    type: CLEAR_JOBS.REQUESTED,
    scope
});

export const getResources = simpleActionCreator(GET_RESOURCES.REQUESTED);
