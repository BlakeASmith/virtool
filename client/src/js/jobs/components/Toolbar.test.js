import * as actions from "../actions";
import JobsToolbar from "./Toolbar";

describe("<JobsToolbar />", () => {
  const initialState = { jobs: { filter: "test" } };
  const store = mockStore(initialState);
  let wrapper;

  it("renders correctly", () => {
    wrapper = shallow(<JobsToolbar store={store} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  describe("Dispatch Functions", () => {
    let spy;

    afterEach(() => {
      spy.restore();
    });

    it("Search input change dispatches filterJobs() action", () => {
      spy = sinon.spy(actions, "filterJobs");
      expect(spy.called).toBe(false);

      wrapper = mount(<JobsToolbar store={store} canRemove={false} />);

      wrapper
        .find({ value: "test" })
        .at(0)
        .prop("onChange")({ target: { value: "search" } });
      expect(spy.calledWith("search")).toBe(true);
    });

    it("Selecting a clear option dispatches clearJobs() action", () => {
      spy = sinon.spy(actions, "clearJobs");
      expect(spy.called).toBe(false);

      wrapper = mount(<JobsToolbar store={store} canRemove={true} />);

      wrapper.find({ eventKey: "removeFailed" }).prop("onClick")({
        target: { name: "failed" }
      });
      expect(spy.calledWith("failed")).toBe(true);
    });
  });
});
