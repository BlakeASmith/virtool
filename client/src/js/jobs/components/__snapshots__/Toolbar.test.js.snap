// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobsToolbar /> renders correctly 1`] = `
<div
  className="toolbar"
>
  <FormGroup
    bsClass="form-group"
  >
    <InputGroup
      bsClass="input-group"
    >
      <InputGroupAddon
        bsClass="input-group-addon"
      >
        <Icon
          faStyle="fas"
          fixedWidth={false}
          name="search"
          pullRight={false}
        />
      </InputGroupAddon>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        placeholder="User or task"
        value="test"
      />
    </InputGroup>
  </FormGroup>
</div>
`;
